src/test/scala/chiselTests/reflect/CheckTypeEquivalenceSpec.scala

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Turns out there were basically zero tests of this API. I did not write any for `Const` nor `Probe` which isn't great, but this needs to backport to 5.x.